plugin/src/main/java/com/redhat/jenkins/nodesharingfrontend/SharedNodeFactory.java

```diff
@@ -51,22 +52,33 @@ public static SharedNode transform(@Nonnull NodeDefinition def) throws IllegalAr
     }
 
     @Nonnull
-    private static SharedNode decorate(@Nonnull SharedNode node) {
-        node.setRetentionStrategy(new SharedOnceRetentionStrategy(2));
+    private static SharedNode decorate(@Nonnull SharedNode node) throws IllegalArgumentException {
+        node.setRetentionStrategy(new SharedOnceRetentionStrategy(1));
+        node.setMode(Node.Mode.EXCLUSIVE);
+        if (node.getNumExecutors() != 1) {
```

Comment:
I don't like forcing like this one. It could make sense to have an opportunity to define more executors even on single box for single run. Also there can be other useful setting in the future which we don't need to support in the plugin/repo conversion directly.

What is the idea behind to force some of the parameters directly and not maintain them through repo definition only? I'm not for feature like this one. If there isn't a demand for specify number of executor or retention strategy lets  remove them from the node definition, that's fine for me.

Reply:
Agreed things we do not support currently (different # of executors, other node modes, etc.) will be disallowed/hardcoded to only known working values to avoid user confusion. Once we hear there are valid uses for changing these we will test, document and relax the restrictions.